Source/JavaScriptCore/API/JSValue.h

```diff
@@ -169,23 +169,60 @@ NS_CLASS_AVAILABLE(10_9, 7_0)
 @param context The JSContext to which the resulting JSValue belongs.
 @result The JSValue representing a unique JavaScript value with type symbol.
 */
-+ (JSValue *)valueWithNewSymbolFromDescription:(NSString *)description inContext:(JSContext *)context JSC_API_AVAILABLE(macos(10.15), ios(13.0));
++ (JSValue * _Null_unspecified)valueWithNewSymbolFromDescription:(NSString * _Null_unspecified)description inContext:(JSContext * _Null_unspecified)context JSC_API_AVAILABLE(macos(10.15), ios(13.0));
+
+/*!
+@method
+@abstract Create a new BigInt value from a numeric string.
+@param string The string representation of the BigInt JavaScript value being created.
+@param context The JSContext to which the resulting JSValue belongs.
+@result The JSValue representing a JavaScript value with type BigInt.
+@discussion This is equivalent to calling the <code>BigInt</code> constructor from JavaScript with a string argument.
+*/
++ (nullable JSValue *)valueWithNewBigIntFromString:(nonnull NSString *)string inContext:(nonnull JSContext *)context JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+@method
+@abstract Create a new BigInt value from a <code>int64_t</code>.
+@param int64 The signed 64-bit integer of the BigInt JavaScript value being created.
+@param context The JSContext to which the resulting JSValue belongs.
+@result The JSValue representing a JavaScript value with type BigInt.
+*/
++ (nullable JSValue *)valueWithNewBigIntFromInt64:(int64_t)int64 inContext:(nonnull JSContext *)context JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+@method
+@abstract Create a new BigInt value from a <code>uint64_t</code>.
+@param uint64 The unsigned 64-bit integer of the BigInt JavaScript value being created.
+@param context The JSContext to which the resulting JSValue belongs.
+@result The JSValue representing a JavaScript value with type BigInt.
+*/
++ (nullable JSValue *)valueWithNewBigIntFromUInt64:(uint64_t)uint64 inContext:(nonnull JSContext *)context JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+@method
+@abstract Create a new BigInt value from a double.
+@param value The value of the BigInt JavaScript value being created.
+@param context The JSContext to which the resulting JSValue belongs.
+@result The JSValue representing a JavaScript value with type BigInt.
+*/
++ (nullable JSValue *)valueWithNewBigIntFromDouble:(double)value inContext:(nonnull JSContext *)context JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
```

Comment:
What happens if the input double value contains a fraction part?  Does this API truncate or round in some way?  Or does it error out and refuse to create a BigInt?  If it errors out, how is the error manifested?  I think this API needs work on clarifying these details.

Reply:
Secondly, what happens if we encounter an OOME while trying to create a BigInt?  How is that represented?  I see that the pre-existing `valueWithNewSymbolFromDescription` didn't specify this either.  Is it specified elsewhere?

Reply:
> What happens if the input double value contains a fraction part? Does this API truncate or round in some way? Or does it error out and refuse to create a BigInt? If it errors out, how is the error manifested? I think this API needs work on clarifying these details.

`valueWithNewBigIntFromDouble` is implemented by using `JSBigIntCreateWithDouble` which has a discussion comment:
```
@discussion If the value is not an integer, an exception is thrown.
```

Yes, we should add that comment for `valueWithNewBigIntFromDouble` too. And it's defined by spec:
https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-bigint-constructor-number-value
https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-numbertobigint
So, no truncation needed for converting from double to bigint as long as the double is an integer.

Reply:
> Secondly, what happens if we encounter an OOME while trying to create a BigInt? How is that represented? I see that the pre-existing valueWithNewSymbolFromDescription didn't specify this either. Is it specified elsewhere?

If any exceptions thrown, the `context` will notify the exception. So, users should check whether the passed `context` got an exception or not after calling `valueWithNewBigIntFromString`. As well as other APIs.